### ISSUE TITLE
Questions about model robustness

### ISSUE THREAD

**Issue by deyang2000** [Traffic-X/MonoUNI#8]
Hello, author!
I enhanced the image of the validation set with a simple rainy & foggy -day simulation program, and it looked like this:

![image](https://github.com/Traffic-X/MonoUNI/assets/129513825/d5165370-62b6-4415-89ac-5079c03524bb)
![image](https://github.com/Traffic-X/MonoUNI/assets/129513825/29403d01-7cd3-4539-8587-87e1d48e32fb)

That is, blur the image, reduce the brightness, and add a mask of some raindrops.
I used the official ckpt for verification, and the comparison is as follows:

********************car********************
bbox@0.70 [96.97587769060773, 99.43916721862426, 97.04305694184455]
bev@0.70 [82.91338665086208, 82.22236411339173, 82.18257252897446]
3d@0.70 [75.5153895932399, 75.27078423603838, 75.1355335795734]
aos [99.27853419324472, 99.37303304437647, 99.33276931153972]
bbox@0.50 [99.45729055841753, 99.5492530942907, 99.51969387573062]
bev@0.50 [91.80759521645166, 92.49802504973516, 92.47992894413947]
3d@0.50 [92.65049690746477, 93.06885181584063, 93.0254487356281]
bev@0.30 [95.01409992120026, 95.54873005854269, 95.53547554066499]
3d@0.30 [96.40101528049155, 96.5902987969818, 96.57004128525296]
2024-05-21 18:22:22,703   
INFO  {'bbox@0.70': [96.97587769060773, 99.43916721862426, 97.04305694184455],
 'bev@0.70': [82.91338665086208, 82.22236411339173, 82.18257252897446],
  '3d@0.70': [75.5153895932399, 75.27078423603838, 75.1355335795734],
   'aos': [99.27853419324472, 99.37303304437647, 99.33276931153972], 
   'bbox@0.50': [99.45729055841753, 99.5492530942907, 99.51969387573062], 
   'bev@0.50': [91.80759521645166, 92.49802504973516, 92.47992894413947], 
   '3d@0.50': [92.65049690746477, 93.06885181584063, 93.0254487356281], 
   'bev@0.30': [95.01409992120026, 95.54873005854269, 95.53547554066499], 
   '3d@0.30': [96.40101528049155, 96.5902987969818, 96.57004128525296]}

********************car********************     rain and fog
bbox@0.70 [86.62723709493258, 84.31233745769502, 81.93751922630769]
bev@0.70 [11.97951735346964, 12.91851984096729, 12.953796092913189]
3d@0.70 [5.602129356343057, 5.8551420736835, 5.849890171969584]
aos [84.02914109959092, 81.92114869458274, 81.86665065264374]
bbox@0.50 [86.99649094524744, 84.62605853292004, 84.59835480523805]
bev@0.50 [39.08887650358005, 40.64569719077104, 40.62059129107645]
3d@0.50 [29.25147127531466, 30.762483848887477, 30.714181702464305]
bev@0.30 [65.57447835387885, 63.23464635696953, 63.1454831542585]
3d@0.30 [59.41559161753111, 59.2939806263183, 59.16924448602208]
 2024-05-21 19:07:44,657   
INFO  {'bbox@0.70': [86.62723709493258, 84.31233745769502, 81.93751922630769], 
'bev@0.70': [11.97951735346964, 12.91851984096729, 12.953796092913189],
 '3d@0.70': [5.602129356343057, 5.8551420736835, 5.849890171969584], 
 'aos': [84.02914109959092, 81.92114869458274, 81.86665065264374], 
 'bbox@0.50': [86.99649094524744, 84.62605853292004, 84.59835480523805], 
 'bev@0.50': [39.08887650358005, 40.64569719077104, 40.62059129107645],
  '3d@0.50': [29.25147127531466, 30.762483848887477, 30.714181702464305], 
  'bev@0.30': [65.57447835387885, 63.23464635696953, 63.1454831542585], 
  '3d@0.30': [59.41559161753111, 59.2939806263183, 59.16924448602208]}

In the original val dataset, all APs of car are relatively high at different IOU thresholds. However, after adding the rain-fog mask, all AP decreased significantly, especially the task metrics involving 3D detection.

I ran a similar experiment on BEVHeight and found it much more robust. Does your approach sacrifice robustness for greater accuracy? And any other possible suggestions? Thank you for your reply.

**Comment by jjrCN** [comment]
This drop point is a bit serious. We have not done similar comparative experiments before. Perhaps adding corresponding data enhancement to the training set will significantly alleviate this phenomenon. After all, the Rope3D data set is also limited in size.
